embulk-input-s3/src/main/java/org/embulk/input/s3/AbstractS3FileInputPlugin.java

```diff
@@ -144,9 +152,46 @@ protected ClientConfiguration getClientConfiguration(PluginTask task)
         clientConfig.setMaxErrorRetry(3); // SDK default: 3
         clientConfig.setSocketTimeout(8*60*1000); // SDK default: 50*1000
 
+        // set http proxy
+        if (task.getHttpProxy().isPresent()) {
+            setHttpProxy(clientConfig, task.getHttpProxy().get());
+        }
+
         return clientConfig;
     }
 
+    private Optional<HttpProxy> setupHttpProxy(PluginTask task)
+    {
+        // If users configure http proxy in in: section, the settings are used. If http proxy settings
+        // don't exist in the config, it searches and extracts from environment variables: "HTTPS_PROXY",
+        // "https_proxy", "HTTP_PROXY", "http_proxy".
+        return task.getHttpProxy().or(HttpProxy.getHttpProxyFromEnv());
+    }
+
+    private void setHttpProxy(ClientConfiguration clientConfig, HttpProxy httpProxy)
```

Comment:
The name of `setHttpProxy` is a bit confusing as the same with the one in `PluginTask`. How about `setHttpProxyInAwsClient(Configuration)`?